lib/bank/checking_account.rb

```diff
@@ -0,0 +1,44 @@
+module Bank
+  class CheckingAccount < Account
+    attr_accessor :check_transactions, :month
+
+    def initialize(id, initial_balance)
+      super
+      @check_transactions = 0
+      @month = Time.now.mon
+    end
+
+    def withdraw(amount)
+      super
+      # there was not an error with the withdrawal amount
+      if !@is_withdrawal_error
+        # subtract the $1 fee from balance
+        return @balance - 1
```

Comment:
You probably want this to be 
```
return @balance =- 1
```
Otherwise, while the method seems to return the right balance, the actual balance will be unchanged (fee not assessed).